File: database-driver-api/src/main/java/de/lino/database/database/exception/DataAlreadyExist.java
package de.lino.database.database.exception;

import org.jetbrains.annotations.NotNull;

/**
 * Thrown when an attempt is made to insert a {@link de.lino.database.database.entity.DatabaseEntry}
 * with an id that already exists in the target {@link de.lino.database.database.DatabaseSection}.
 */
public class DataAlreadyExist extends RuntimeException {

    /**
     * Creates a new exception for the entry with the given, already existing, id.
     *
     * @param id the primary key that was already present
     */
    public DataAlreadyExist(@NotNull String id) {
        super("Entry already exists with id='" + id + "'");
    }

}

File: python/database-driver-api/src/database_driver/api/database/exception/data_already_exist.py
"""Mirror of ``de.lino.database.database.exception.DataAlreadyExist``."""

from __future__ import annotations


class DataAlreadyExist(RuntimeError):
    """Raised when an attempt is made to insert a ``DatabaseEntry`` with an id that
    already exists in the target ``DatabaseSection``."""

    def __init__(self, id: str) -> None:
        super().__init__(f"Entry already exists with id='{id}'")
